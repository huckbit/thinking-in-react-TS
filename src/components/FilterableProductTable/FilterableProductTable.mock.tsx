import type { Product } from "../ProductRow/ProductRow";

interface Products {
  products: Product[];
}
const mock: Products = {
  products: [
    { category: "Fruits", price: "£1", stocked: true, name: "Apple" },
    { category: "Fruits", price: "£1", stocked: true, name: "Dragonfruit" },
    { category: "Fruits", price: "£2", stocked: false, name: "Passionfruit" },
  ],
};

export default mock;
