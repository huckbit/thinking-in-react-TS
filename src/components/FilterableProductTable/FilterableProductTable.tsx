import { useState } from "react";
import ProductTable, { ProductTableProps } from "../ProductTable";
import type { Product } from "../ProductRow";

import SearchBar from "../SearchBar";

function FilterableProductTable({ products }: { products: Product[] }) {
  const [filterText, setFilterText] = useState<string>("");
  const [inStockOnly, setInStockOnly] = useState<boolean>(false);

  return (
    <div>
      <SearchBar
        filterText={filterText}
        inStockOnly={inStockOnly}
        onInStockOnlyChange={setInStockOnly}
        onFilterTextChange={setFilterText}
      />
      <ProductTable products={products} filterText={filterText} inStockOnly={inStockOnly} />
    </div>
  );
}

export default FilterableProductTable;
