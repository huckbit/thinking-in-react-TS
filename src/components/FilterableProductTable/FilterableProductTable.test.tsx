/// <reference types="@testing-library/jest-dom" />
import { render } from "@testing-library/react";
import FilterableProductTable from "./FilterableProductTable";
import mock from "./FilterableProductTable.mock";

test("matches snapshot", () => {
  const { asFragment } = render(<FilterableProductTable products={mock.products} />);
  expect(asFragment()).toMatchSnapshot();
});
