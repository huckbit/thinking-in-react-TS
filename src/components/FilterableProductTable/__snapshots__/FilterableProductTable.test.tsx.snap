// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        placeholder="Search..."
        type="text"
        value=""
      />
      <label>
        <input
          type="checkbox"
          value="false"
        />
        Only show products in stock
      </label>
    </form>
    <table>
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Price
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th
            colspan="2"
          >
            Fruits
          </th>
        </tr>
        <tr>
          <td>
            Apple
          </td>
          <td>
            £1
          </td>
        </tr>
        <tr>
          <td>
            Dragonfruit
          </td>
          <td>
            £1
          </td>
        </tr>
        <tr>
          <td>
            <span
              style="color: red;"
            >
              Passionfruit
            </span>
          </td>
          <td>
            £2
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
